src/types.ts

```diff
@@ -11,8 +11,9 @@ export type TrustResolution = {
 }
 
 export type ResolverConfig = {
-  trustRegistryUrl: string
+  trustRegistryUrl?: string
   didResolver?: Resolver
+  agent?: AgentContext
```

Comment:
```suggestion
  agentContext: AgentContext
```

Use `agentContext` for all variables of type AgentContext to make it clear that we'll use a context rather than an Agent instance.

This parameter must be mandatory. Otherwise it won't work.